stl/inc/syncstream

```diff
@@ -0,0 +1,330 @@
+// syncstream standard header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _SYNCSTREAM_
+#define _SYNCSTREAM_
+#include <yvals.h>
+#if _STL_COMPILER_PREPROCESSOR
+#define __cpp_lib_syncbuf // Remove before commit, visual
+#ifndef __cpp_lib_syncbuf
+#pragma message("The contents of <syncstream> are available only with C++20 syncbuf support.")
+#else // ^^^ !defined(__cpp_lib_syncbuf) / defined(__cpp_lib_syncbuf) vvv
+#include <map>
+#include <memory>
+#include <mutex>
+#include <streambuf>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#pragma push_macro("emit")
+#undef new
+#undef emit
+
+_STD_BEGIN
+struct _Basic_syncbuf_global_mutex {
+private:
+    struct _Mutex_count_pair {
+        mutex _Mutex;
+        uint64_t _Ref_count = 0;
+    };
+    static map<void*, _Mutex_count_pair>& _Get_mutex_map() noexcept {
+        static map<void*, _Mutex_count_pair> _Mutex_Map;

```

Comment:
We shouldn't use function scope static variables, so this and map mutex should be brought to namespace scope (after that these functions can be removed). They might have to be built into a DLL but I'm not sure about that.

Reply:
I get from your review, that we should allow these functions to exist directly in `::std` namespace. And to achieve thread-safety, the map could be in a DLL. I am wondering how I can add the variables in the DLL "not sure how it works for STL 😕"

Reply:
Unfortunately I also don't know it so we need to ask a maintainer.